src/utils/Api.js

```diff
@@ -0,0 +1,126 @@
+class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  getCurrentUser() {
+    return fetch(`${this._baseURL}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    })
+      .then((res) => {
+        return this._checkResponse(res);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.